### PR TITLE
8333560: -Xlint:restricted does not work with --release

### PR DESCRIPTION
**Pull request by JornVernee**
Currently, `-Xlint:restricted` can be used to print out lint warnings when restricted methods are used in source code compiled by javac. However, this option doesn't work as intended when used in combination with `--release`, and instead no warnings are printed at all.

Javac relies on the internal `jdk.internal.javac.Restricted` annotation being present on a method in order to determine if a method is restricted. This annotation is however currently being stripped out when generating shim class files that are used by javac to get release-specific symbol information.

There are already some annotations that are exempt from being stripped away, such as `jdk.internal.javac.PreviewFeature`. The proposed fix is to add `jdk.internal.javac.Restricted` to the list of exempted annotations as well.

Testing: tier 1-3

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333560](https://bugs.openjdk.org/browse/JDK-8333560): -Xlint:restricted does not work with --release (**Bug** - P3)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**) ⚠️ Review applies to [9dfaf070](https://git.openjdk.org/jdk/pull/19548/files/9dfaf070de6850391b2485bb32a95d57a98fbced)
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - **Reviewer**) ⚠️ Review applies to [cadc668b](https://git.openjdk.org/jdk/pull/19548/files/cadc668be29ad368019ce30514fb2a5f621db678)

### Contributors
 * Jan Lahoda `<jlahoda@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19548/head:pull/19548` \
`$ git checkout pull/19548`

Update a local copy of the PR: \
`$ git checkout pull/19548` \
`$ git pull https://git.openjdk.org/jdk.git pull/19548/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19548`

View PR using the GUI difftool: \
`$ git pr show -t 19548`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19548.diff">https://git.openjdk.org/jdk/pull/19548.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19548#issuecomment-2149288039)

### CODE REVIEW

**Comment by JornVernee** [comment]
Spoke to @lahodaj offline, and he pointed out that we need to add a fake annotation for `@Restricted` as well. Javac will use the fake annotation to set the internal `RESTRICTED` flag, but it is not otherwise exposed.

/contributor add @lahodaj

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Comment by JornVernee** [comment]
I'm taking a look at the failed  test. Will re-run tier 1-3 as well.

**Comment by vicente-romero-oracle** [comment]
> I'm taking a look at the failed test. Will re-run tier 1-3 as well.

what test failed? just curious

**Comment by JornVernee** [comment]
> > I'm taking a look at the failed test. Will re-run tier 1-3 as well.
> 
> what test failed? just curious

`tools/javac/processing/model/TestSymtabItems.java` failed. We need to use `enterSyntheticAnnotation` in `Symtab` instead of `enterClass` I think.

**Comment by lahodaj** [comment]
> > > I'm taking a look at the failed test. Will re-run tier 1-3 as well.
> > 
> > 
> > what test failed? just curious
> 
> `tools/javac/processing/model/TestSymtabItems.java` failed. We need to use `enterSyntheticAnnotation` in `Symtab` instead of `enterClass` I think.

Oops, yes, sorry.

File: make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java [hunk 1]
```diff
@@ -338,6 +338,10 @@ public void createSymbols(String ctDescriptionFileExtra, String ctDescriptionFil
             "Ljdk/internal/PreviewFeature;";
     private static final String PREVIEW_FEATURE_ANNOTATION_NEW =
             "Ljdk/internal/javac/PreviewFeature;";
+    private static final String RESTRICTED_ANNOTATION =
+            "Ljdk/internal/javac/Restricted;";
+    private static final String RESTRICTED_ANNOTATION_INTERNAL =
```
Comment (mcimadamore) [inline make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java:343]: These should be moved around, now we're intermixing preview-related decls with restricted ones

**Review by mcimadamore — APPROVED** [review]

**Pull request closed without merging**